### PR TITLE
[5.3] Changing the name of the method mapToAssoc() to mapWithKeys()

### PR DESCRIPTION
**Pull request by fernandobandeira**
This was suggested in #14795 by @JosephSilber and @adamwathan and is a better definition of this method.

Also changed the test accordingly.

### CODE REVIEW

File: src/Illuminate/Support/Collection.php [hunk 1]
```diff
@@ -602,12 +602,12 @@ public function map(callable $callback)
     /**
      * Run an associative map over each of the items.
      *
-     * The callback should return an associative array with a single key/value pair.
+     * Map a collection persisting any changes made to keys.
```
Comment (JosephSilber) [inline src/Illuminate/Support/Collection.php]: Why did you change this?

Comment (fernandobandeira) [inline src/Illuminate/Support/Collection.php]: Whoops changed the wrong line, it was meant for the 'Run an associative map over each of the items.'

**Comment by tomschlick** [comment]
There should probably be a fallback method with the old name here as to not introduce a breaking change. Just add a deprecated docblock on top of it so people are aware it will go away at some point.

**Comment by fernandobandeira** [comment]
I think it's not needed since this method were introduced last week on 5.3 only so it shouldn't be a BC since 5.3 isn't even out yet.

**Comment by tomschlick** [comment]
Ah, didn't know that it was introduced only last week. Carry on! 😄

**Comment by GrahamCampbell** [comment]
RC1 is out now. We maybe need to bit a little bit careful at this point. Nothing's set in stone though I think. :)
